core/src/main/scala/org/apache/spark/scheduler/TaskResultGetter.scala

```diff
@@ -64,6 +64,8 @@ private[spark] class TaskResultGetter(sparkEnv: SparkEnv, scheduler: TaskSchedul
           val (result, size) = serializer.get().deserialize[TaskResult[_]](serializedData) match {
             case directResult: DirectTaskResult[_] =>
               if (!taskSetManager.canFetchMoreResults(serializedData.limit())) {
+                scheduler.handleFailedTask(taskSetManager, tid, TaskState.KILLED, TaskKilled(
```

Comment:
Better to leave a comment here to explain why we handle the oversize task as a killed task.

Reply:
Updated, thanks.